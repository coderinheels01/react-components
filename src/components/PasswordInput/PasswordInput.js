import React, { useState } from "react";
import PropTypes from "prop-types";
import TextInput from "../TextInput";
import EyeIcon from "../EyeIcon";
import ProgressBar from "../ProgressBar";
import { Link } from "react-router-dom";

/** Password input with show/hide password icon*/
const PasswordInput = ({
  label = "Password",
  value,
  minLength = 8,
  placeholder,
  htmlId,
  quality,
  showTogglePasswordIcon = false,
  name,
  error,
  onChange
}) => {
  const [showPassword, setShowPassword] = useState(false);

  return (
    <div>
      <TextInput
        label={label}
        type={showPassword ? "text" : "password"}
        value={value}
        minLength={minLength}
        placeholder={placeholder}
        htmlId={htmlId}
        name={name}
        onChange={onChange}
        error={error}
        required
      >
        {showTogglePasswordIcon && (
          <Link onClick={() => setShowPassword(!showPassword)}>
            <EyeIcon />
          </Link>
        )}
        <ProgressBar percent={quality} width={150} height={8} />
      </TextInput>
    </div>
  );
};

PasswordInput.propTypes = {
  /** Label text of the password input box*/
  label: PropTypes.string,

  /** Password value */
  value: PropTypes.any,

  /** Maximum length of password input box*/
  minLength: PropTypes.number,

  /** Default message if Input is empty */
  placeholder: PropTypes.string,

  /**HTML ID for the input */
  htmlId: PropTypes.string,

  /** ProgressBar percentage to display quality of the password*/
  quality: PropTypes.number,

  /** Flag to show or hide eye icon to show/hide visibility of password*/
  showTogglePasswordIcon: false,

  /** Password Input name */
  name: PropTypes.string,

  /** Error string to display */
  error: PropTypes.node,

  /** Function to call when input value changes*/
  onChange: PropTypes.func.isRequired
};

export default PasswordInput;
