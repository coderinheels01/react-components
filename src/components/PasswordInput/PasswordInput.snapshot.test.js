import React from "react";
import renderer from "react-test-renderer";
import PasswordInput from "./PasswordInput";

describe("PasswordInput", () => {
  let passwordInputTree;
  let props;
  describe("render PasswordInput without EyeIcon", () => {
    beforeEach(() => {
      props = {
        value: "test value",
        minLength: 8,
        placeholder: "test placeholder",
        htmlId: "test-html-id",
        quality: 20,
        name: "test name",
        error: "some error",
        onChange: jest.fn()
      };
      passwordInputTree = renderer.create(<PasswordInput {...props} />);
    });

    it("should render PasswordInput without showPassword icon", () => {
      expect(passwordInputTree).toMatchSnapshot();
    });

    afterEach(() => {
      passwordInputTree.unmount();
    });
  });
});
