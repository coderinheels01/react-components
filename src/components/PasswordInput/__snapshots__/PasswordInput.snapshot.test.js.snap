// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput render PasswordInput without EyeIcon should render PasswordInput without showPassword icon 1`] = `
<div>
  <div
    className="sc-dlfnbm InidV"
  >
    <label
      className="sc-bdfBwQ gaSFkq"
      htmlFor="test-html-id"
    >
      Password
      <span
        className="sc-gsTCUz wIMML"
      >
         * 
      </span>
    </label>
    <input
      className="sc-hKgILt brxkeH"
      id="test-html-id"
      minLength={8}
      name="test name"
      onChange={[MockFunction]}
      placeholder="test placeholder"
      type="password"
    />
    <div
      className="sc-jSgupP clBKku"
      width={150}
    >
      <div
        className="sc-gKsewC eULEFe"
        height={8}
        width={30}
      />
    </div>
    <div
      className="sc-eCssSg dIgmt"
    >
      some error
    </div>
  </div>
</div>
`;

exports[`PasswordInput when showTogglePasswordIcon is true or undefined should render PasswordInput without showPassword icon 1`] = `
<div>
  <div
    className="sc-dlfnbm InidV"
  >
    <label
      className="sc-bdfBwQ gaSFkq"
      htmlFor="test-html-id"
    >
      Password
      <span
        className="sc-gsTCUz wIMML"
      >
         * 
      </span>
    </label>
    <input
      className="sc-hKgILt brxkeH"
      id="test-html-id"
      minLength={8}
      name="test name"
      onChange={[MockFunction]}
      placeholder="test placeholder"
      type="password"
    />
    <div
      className="sc-jSgupP clBKku"
      width={150}
    >
      <div
        className="sc-gKsewC eULEFe"
        height={8}
        width={30}
      />
    </div>
    <div
      className="sc-eCssSg dIgmt"
    >
      some error
    </div>
  </div>
</div>
`;
